### PR TITLE
Remove shops from multistore header shop list if they are not associated with current employee

### LINKED ISSUE #27377: multistore_header appears, even if employees has no right on multi shop
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

I'm on a multishop
I create a employee, with access only to one shop.

On several pages, he see the banner multishop, with all shops available.
He can't access to them, but see their name and url.

### Expected behavior

The multistore_header should not appears if the employee has not right on multishop

### Steps to reproduce

1. Create a multistore
2. Create on employee, with access to only one shop.
3. Connect to this employee account.
4. You will see the multishop header, on all page where it's available (orders, for example)

### PrestaShop version(s) where the bug happened

1.7.8.0

### PHP version(s) where the bug happened

7.4

### If your bug is related to a module, specify its name and its version

_No response_

### LINKED ISSUE #27704: Multistore settings accessible by non-permissioned team members
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

https://youtu.be/UtOdXzM1Tnk

When clicking "edit" in the multistore-header menu it is possible to go into the settings of the multistore for un-permissioned users. 

### Expected behavior

un-permissioned users should not see the edit button and not be able to go into multistore settings. 

### Steps to reproduce

SEE VIDEO
https://youtu.be/UtOdXzM1Tnk

1. create multistore
2. create team member "example"
3. create profile "test"
4. customize profile "test" with only product permissions
5. assign team member "example" profile "test"
6. sign into multistore backoffice with user "example"
7. click on the multistore-header edit button
8. get access to multistore settings ...

### PrestaShop version(s) where the bug happened

1.7.8.1

### PHP version(s) where the bug happened

7.4.25

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by alisamie97**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 1.7.8.x
| Description?      | Check issues descriptions. When you create a new employee that is associated to a list of specific shops and not all of them, you expect that employee has access to those shops only and not all the shops. But if you login with the new employee account, you can see in the multistore header that all shops are available. This PR is a fix for this issue.
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes  #27704 and #27377
| Related PRs       | No
| How to test?      | Enable multistore and create an extra shop. Create a new profile other in BO > Configure > Advanced Parameters > Team > Profiles. Then create give all permissions to that profile. And finally create a new employee with that profile and associate to the new shop you just created. Do not associate to the main shop. Then login via that new employee account and check multistore header. You should see only the associated shops in the create employee form that you have selected. You can also verify this by manually altering the `ps_employee_shop` table
| Possible impacts? | Not sure

### CODE REVIEW

**Comment by alisamie97** [comment]
Keep in mind that if you select **_Super Admin_** as the new employee profile in new employee form, you are associating all the shops with this new employee by default and blindly. Just check this:
https://github.com/PrestaShop/PrestaShop/blob/9887e7ece9810d847b8f31558ac2baabd1a91659/src/Core/Form/IdentifiableObject/DataHandler/EmployeeFormDataHandler.php#L136-L139

File: src/PrestaShopBundle/Controller/Admin/MultistoreController.php [hunk 1]
```diff
@@ -93,6 +93,17 @@ public function header(bool $lockedToAllShopContext): Response
             $groupList = $this->entityManager->getRepository(ShopGroup::class)->findBy(['active' => true]);
         }
 
+        $associatedShops = $this->getContext()->employee->getAssociatedShops();
+
+        foreach ($groupList as $group) {
+            $group = $group;
```
Comment (alisamie97) [inline src/PrestaShopBundle/Controller/Admin/MultistoreController.php]:
```suggestion
```

**Comment by alisamie97** [comment]
Can somebody tell me what is wrong with the code that cs fixer is showing me?

**Comment by matks** [comment]
> Can somebody tell me what is wrong with the code that cs fixer is showing me?

A simple blank space is needed after `if`
Capture d’écran 2022-08-17 à 15 28 48

**Comment by kpodemski** [comment]
Hi @stifler97 

I think you could use cs-fixer locally:
https://devdocs.prestashop.com/1.7/modules/testing/basic-checks/#coding-standards

I highly recommend you check how the CI works in PrestaShop and how it helps us maintain the codebase's consistency.

File: src/PrestaShopBundle/Controller/Admin/MultistoreController.php [hunk 2]
```diff
@@ -93,6 +93,16 @@ public function header(bool $lockedToAllShopContext): Response
             $groupList = $this->entityManager->getRepository(ShopGroup::class)->findBy(['active' => true]);
         }
 
+        $associatedShops = $this->getContext()->employee->getAssociatedShops();
```
Comment (matks) [inline src/PrestaShopBundle/Controller/Admin/MultistoreController.php:96]: Interesting. I am worried about one usecase: is it possible to configure an employee to "have access to ALL SHOPS" and if yes, does this code work with it?

I think this code works
- when an employee is configured to have access to some shops in particular (example: employee Mathieu F has access to shops 1, 4 and 5 but not 2 and 3)
- when an employee is configured to have access to some group of shops in particular (example: employee Mathieu F has access to groups 1 and 3 but not 2)

I am worried about the 3rd usecase

**Comment by MatShir** [comment]
Hi @stifler97, we are planning a patch release for September, we would like to have your contribution in it. Have you checked @matks comment ?

File: src/PrestaShopBundle/Controller/Admin/MultistoreController.php [hunk 3]
```diff
@@ -93,6 +93,16 @@ public function header(bool $lockedToAllShopContext): Response
...
+
+        foreach ($groupList as $group) {
+            foreach ($group->getShops() as $key => $shop) {
+                if(!in_array($shop->getId(), $associatedShops)) {
```
Comment (alisamie97) [inline src/PrestaShopBundle/Controller/Admin/MultistoreController.php]:
```suggestion
                if (!in_array($shop->getId(), $associatedShops)) {
```

File: src/PrestaShopBundle/Controller/Admin/MultistoreController.php (hunk 2 above)
Comment (alisamie97) [reply src/PrestaShopBundle/Controller/Admin/MultistoreController.php:96]: I did not see any code that checks shop groups so there is no worries for groups. But it is a nice feature to have.

**Comment by alisamie97** [comment]
> Hi @stifler97, we are planning a patch release for September, we would like to have your contribution in it. Have you checked @matks comment ?

Hi. That is a good news for me as a new comer. I just added a new fix to this PR.

File: src/PrestaShopBundle/Controller/Admin/MultistoreController.php (hunk 2 above)
Comment (yanmakouf) [reply src/PrestaShopBundle/Controller/Admin/MultistoreController.php:96]: https://github.com/PrestaShop/PrestaShop/pull/29342#issuecomment-1215145255 As you mentioned here, I think the 3rd case @matks is worried about is covered by this line. If an employee is super admin and has access to all shops then `getAssociatedShops ()` will simply return all ids. So it looks ok to me, but maybe worth testing the third case :)

**Review by yanmakouf — APPROVED** [review]

**Comment by alisamie97** [comment]
Hi every one. Is every thing ok with this PR? Just a reminder.

**Review by atomiix — APPROVED** [review]

**Comment by alisamie97** [comment]
just a reminder. It has been a month since I opened this PR

**Comment by samberrry** [comment]
Hey @stifler97, 
I thought that it would be better to familiarize you with this space more. 

In short, you must be patient. Adding something one month or two months ago does not mean that it should be merged immediately. Normally, Ps team manage and prioritize tasks within projects, please take a look at:
https://github.com/PrestaShop/PrestaShop/projects/26#card-85029280 
Ps is a large project with many tasks with different aspects, so they know better when it should be added, and to which version. However, you can track your contribution through the projects and issues. Take a look at the issue: https://github.com/PrestaShop/PrestaShop/issues/27377
you can see that it has been labeled many times (take a look at this cart too: https://github.com/PrestaShop/PrestaShop/projects/26#card-85029280)

So, everything is normal ;)

**Review by kpodemski — APPROVED** [review]

**Comment by kpodemski** [comment]
Thank you, @stifler97, for your PR; I've just approved it and assigned it to the QA team.

Thanks, @samberrry, for your help. The truth is that this PR was lost because we have a problem tracking PRs with only one remaining review.

**Review by HanaRebaiQA — COMMENTED** [review]
hello @stifler97 

I have checked this PR. The related issues seem not to be fixed.

- Issue https://github.com/PrestaShop/PrestaShop/issues/27377 : NOK

https://user-images.githubusercontent.com/92912932/199017855-b9f86092-fd13-47be-8280-ae66ab89cd8d.mp4



- Issue https://github.com/PrestaShop/PrestaShop/issues/27704 : NOK 

https://user-images.githubusercontent.com/92912932/199017849-44ef2232-1f2c-4f51-a733-d5b132e547a5.mp4

Could you please check again?


Thank You!

**Review by kpodemski — CHANGES_REQUESTED** [review]
Hello @stifler97 

1.7.8.x doesn't allow new improvements and features anymore. Would you be interested in moving your improvement to the `develop` branch and addressing QA feedback?

**Comment by stifler97** [comment]
> 

Hi @kpodemski Yes I want to push this edit to develop. Do I need to close this pr and try a new one?

**Pull request closed without merging**